### PR TITLE
ANA-686 add ZeroCouponBond example.

### PR DESCRIPTION
**Pull request by zvyagin1**
# Pull Request Checklist

- [x] Read the [contributing guidelines](../blob/master/docs/CONTRIBUTING.md)
- [x] Tests pass
- [x] Raised the PR against the `develop` branch

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

I added an example how ZeroCouponBond can be created and used.

### CODE REVIEW

File: sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs [hunk 1]
```diff
@@ -87,6 +87,27 @@ internal static LusidInstrument CreateExampleBond()
...
+            => new Bond(
+                startDate: new DateTimeOffset(2020, 2, 7, 0, 0, 0, TimeSpan.Zero),
+                maturityDate: new DateTimeOffset(2020, 9, 18, 0, 0, 0, TimeSpan.Zero),
+                domCcy: "USD",
+                principal: 100m,
+                couponRate: 0m,
+                flowConventions: 
+                    new FlowConventions(
+                        currency: "USD",
+                        paymentFrequency: "0Invalid",
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs:100]: what does this mean? where does a user get a list of fields that they can use? we need to add either comments for them to resolve or links to a KB article

File: sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs [hunk 2]
```diff
@@ -87,6 +87,27 @@ internal static LusidInstrument CreateExampleBond()
                 instrumentType: LusidInstrument.InstrumentTypeEnum.Bond
             );
 
+        internal static LusidInstrument CreateExampleZeroCouponBond()
+            => new Bond(
+                startDate: new DateTimeOffset(2020, 2, 7, 0, 0, 0, TimeSpan.Zero),
+                maturityDate: new DateTimeOffset(2020, 9, 18, 0, 0, 0, TimeSpan.Zero),
+                domCcy: "USD",
+                principal: 100m,
```
Comment (rizwansaeed) [inline sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs:95]: what units and cost should transactions/holdings be booked with (its not clear from this example)?

File: sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs (hunk 1 above)
Comment (zvyagin1) [reply sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs:100]: Unfortunetelly right now this is the only way how a ZCB schedule can be created. `0Invalid` is 

- not documented
- will be replaced in future

File: sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs (hunk 2 above)
Comment (zvyagin1) [reply sdk/Lusid.Sdk.Tests/Utilities/InstrumentExamples.cs:95]: The ZCB creation is more-or-less identical to a bond creation example we already had before. The only _difference_ is that there is a single payment and coupon rate is zero. The asked question (if clarification is requited) should not be answered in the ZCB example, as other instruments use the same  **domCcy**, **principal** parameters.

**Pull request merged**